### ISSUE TITLE
Possible null value for SynchronizationContext

### ISSUE THREAD

**Issue by rubend30** [MintPlayer/MintPlayer.Dotnet.Tools#5]
In the method `RunOnMainThread `I had a problem where the `synchronizationContext `was `null`. Can this be handled correctly? For now I added a `null` check in a local copy of the library.

```
        private void RunOnMainThread<TState>(Action<TState> action, TState state)
        {
            // added the null check
            if (synchronizationContext == SynchronizationContext.Current || synchronizationContext is null)
            {
                action(state);
            }
            else if(synchronizationContext != null)
            {
                synchronizationContext.Send((param) => action((TState)param), state);
            }
        }
```

**Comment by PieterjanDeClippel** [comment]
Hi,

Thanks for reporting the missing null-check.
I've published a new version with the fix just now.

Would it be possible to check if the issue is resolved with the latest version v1.3.0?
The issue will remain open so you can always report back until the problem is actually solved.

-------------

You can always create a pull-request to update the code of this package. To do so:
- Fork the repository using the button on top-right
- This will create a copy of the repository on your own user account
- Checkout your repository `git clone https://github.com/[username]/MintPlayer.ObservableCollection`
- Go to the project folder `cd MintPlayer.ObservableCollection`
- Create a new branch where you will commit your changes `git branch my-new-feature`
- Checkout your new branch `git checkout my-new-feature`. Every commit you'll do from now will end up on this branch
- Modify the code
- Add your changes+files - `git add .`
- Commit your changes `git commit -m "My new feature"`. This is only on your local git repository.
- Checkout the master branch again `git checkout master`
- 2 possibilities
    - If you get the message `Your branch is up to date with 'origin/master'.`
        - you can switch back to your branch `git checkout my-new feature`
    - If you get another message, then there are commits on `origin/master` you don't have locally, and you need to pull those
        - Pull commits from `origin/master` - `git pull`
        - Checkout your branch again - `git checkout my-new-feature`
        - Merge new commits from the master into your branch - `git merge master`
- Push the new branch with its commits to your remote repository `git push -u origin my-new-feature`
- Go to https://github.com/[username]/MintPlayer.ObservableCollection
- Normally Github should already display a big notification on top with a button **Compare & pull request**
    - If this is not the case:
    - Switch to your branch using the dropdown on the left
    - There's a notification **This branch is x commits ahead of xxx:master**
    - Below the green "Download code" button there's **Contribute - Open pull request**
- The initial setup should display correctly. But here you should select that you want to merge **your** new branch in your repository (second row) into the **master** branch in **my** repository (first row)
    - When you scroll down you should see a diff containing the changes you made
    - Click the **Create pull request** button
- Now it's up to me to accept the changes

**Comment by rubend30** [comment]
Hey

This seems to have fixed the issue. I will try working more with pull request when I know how to fix an issue (kinda new to the whole GitHub ecosystem)

Thank you!
